stl/inc/bit

```diff
@@ -151,24 +139,65 @@ _NODISCARD int _Checked_x86_x64_countl_zero(const _Ty _Val) noexcept {
         const unsigned int _High = _Val >> 32;
         const auto _Low          = static_cast<unsigned int>(_Val);
         if (_High == 0) {
-            return 32 + _Checked_x86_x64_countl_zero(_Low);
+            return 32 + _Countl_zero_lzcnt(_Low);
         } else {
-            return _Checked_x86_x64_countl_zero(_High);
+            return _Countl_zero_lzcnt(_High);
         }
 #else // ^^^ _M_IX86 / !_M_IX86 vvv
         return static_cast<int>(__lzcnt64(_Val));
 #endif // _M_IX86
     }
-    // note: we don't need to call a fallback here because
-    // all supported x86 processors at least have bsr/bsf
+}
+
+template <class _Ty>
+_NODISCARD int _Countl_zero_bsr(const _Ty _Val) noexcept {
+    constexpr int _Digits = numeric_limits<_Ty>::digits;
+
+    unsigned long _Result;
+    if constexpr (_Digits <= 32) {
+        if (!_BitScanReverse(&_Result, _Val)) {
+            return _Digits;
+        }
+    } else {
+#ifdef _M_IX86
+        const unsigned int _High = _Val >> 32;
+        if (_BitScanReverse(&_Result, _High)) {
+            return static_cast<int>(31 - _Result);
+        }
+
+        const auto _Low = static_cast<unsigned int>(_Val);

```

Comment:
it would be good to explicitly make sure we're testing this (the bsr version). Perhaps with just an explicit call in the test code

Reply:
I think this would merit an exception to our usual policy of avoiding testing internal machinery. If you want to add such testing, remember that it should be guarded for x86/x64 like the internal helper.

Reply:
Filed as #1120, thanks.